### PR TITLE
[10.0][MIG] crm_location

### PR DESCRIPTION
**Pull request by luismontalba**
CRM location
-----------------
This module introduces a better zip.
It enables zip, city, state and country auto-completion on lead.

cc @Tecnativa

### CODE REVIEW

**Comment by luismontalba** [comment]
Travis is failing due to this module depends on base_location located in OCA/partner-contact repo.

**Comment by pedrobaeza** [comment]
It shouldn't fail, because you can specify on oca_repositories.txt these kind of dependencies.

File: crm_location/models/crm_lead.py [hunk 1]
```diff
@@ -45,11 +26,8 @@ def on_change_city(self):
     )
 
     @api.multi
-    def on_change_partner_id(self, partner_id):
-        res = super(CrmLead, self).on_change_partner_id(partner_id)
-        if 'value' not in res:
-            res['value'] = {}
+    def _onchange_partner_id_values(self, partner_id):
```
Comment (pedrobaeza) [inline crm_location/models/crm_lead.py]: This old onchange doesn't work for sure. Please use new API onchange (you can create a new method for not having to overwrite super):
```
@api.onchange('partner_id')
def onchange_partner_id_crm_location(self):
    if self.partner_id:
        self.location_id = self.partner_id.zip_id.id
```

File: crm_location/tests/test_crm_location.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        self.assertEqual(lead.state_id, location.state_id)
+        self.assertEqual(lead.country_id, location.country_id)
+
+    def test_onchange_partner_id_values(self):
+        lead = self.lead
+        partner = self.partner
+        location = self.location
+        partner.zip_id = location.id
+        lead.partner_id = partner.id
+        lead._onchange_partner_id_values(lead.partner_id.id)
```
Comment (pedrobaeza) [inline crm_location/tests/test_crm_location.py]: Call the new onchange method.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
And squash your commits together

**Comment by luismontalba** [comment]
Changes proposed done and own commits squashed.

File: crm_location/__manifest__.py [hunk 3]
```diff
@@ -1,39 +1,20 @@
...
     'category': 'Customer Relationship Management',
-    'version': '8.0.1.0.0',
+    'version': '10.0.1.0.0',
     'depends': [
         'crm',
         'base_location',
     ],
     'data': ['views/crm_lead_view.xml'],
-    'author': 'Antiun Ingeniería S.L., '
+    'author': 'Antiun Ingeniería, '
```
Comment (yajo) [inline crm_location/__manifest__.py]: Add Tecnativa

**Review by yajo — APPROVED** [review]

File: crm_location/__manifest__.py (hunk 3 above)
Comment (pedrobaeza) [reply crm_location/__manifest__.py]: Better to replace, as this version is not maintained anymore by these old companies

**Comment by luismontalba** [comment]
Author replaced, new commit squashed.

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**